OpenRA.Mods.Cnc/Activities/Infiltrate.cs

```diff
@@ -40,8 +41,8 @@ protected override void OnInside(Actor self)
 			if (!infiltrates.Info.ValidStances.HasStance(stance))
 				return;
 
-			if (cloak != null && cloak.Info.UncloakOn.HasFlag(UncloakType.Infiltrate))
-				cloak.Uncloak();
+			foreach (var ini in notifiers)
+				ini.Infiltrating(self);
 
 			foreach (var t in target.TraitsImplementing<INotifyInfiltrated>())
```

Comment:
There is already a very similar interface so I suggest to either use that one or move the `Infiltrating` check before `OnInside` is triggered.

Reply:
This is using the traits of the `target`, while we want the traits of `self`. Moving could work, but are you sure that we don't regress anything else then? (The uncloaking also happened *after* checking if the infiltration is valid.)

Reply:
Oh, I see.